xdsl/dialects/vector.py

```diff
@@ -128,7 +130,9 @@ def get(source: Operation | SSAValue) -> BroadcastOp:
 class FMAOp(IRDLOperation):
     name = "vector.fma"
 
-    T: ClassVar = VarConstraint("T", base(VectorType))
+    T: ClassVar = VarConstraint(
+        "T", ParamAttrConstraint(VectorType, (AnyAttr(), AnyFloatConstr, AnyAttr()))
+    )
```

Comment:
Probably best for @math-fehr to merge this first, then use that here/

https://github.com/xdslproject/xdsl/pull/4191/files#diff-b8dee9509fcf97e7a54db20b56724fbe2905745d98c46d0e54547672203e198aR1093-R1108